src/dotnet/CarbonAware.Plugins.JsonReaderPlugin/CarbonAwareJsonReaderPlugin.cs

```diff
@@ -0,0 +1,80 @@
+﻿using System.Collections;
+using CarbonAware.Model;
+using Microsoft.Extensions.Logging;
+using System.Reflection;
+using Newtonsoft.Json;
+
+namespace CarbonAware.Plugins.JsonReaderPlugin;
+
+public class CarbonAwareJsonReaderPlugin : ICarbonAware
+{
+    private readonly ILogger<CarbonAwareJsonReaderPlugin> _logger;
+
+
+    public CarbonAwareJsonReaderPlugin(ILogger<CarbonAwareJsonReaderPlugin> logger)
+    {
+        _logger = logger ?? throw new ArgumentNullException(nameof(logger));
+    }
+
+    public async Task<IEnumerable<EmissionsData>> GetEmissionsDataAsync(IDictionary props)
+    {
+        var data = GetSampleJson();
+
+        return await Task.Run(() => GetFilteredData(data, props));
+    }
+
+    private IEnumerable<EmissionsData> GetFilteredData(IEnumerable<EmissionsData> data, IDictionary props) {
+        var l = props[CarbonAwareConstants.Locations] as IEnumerable<string>;
```

Comment:
Can we use more descriptive variable names here and elsewhere?

Reply:
Fixed the variable names. As far checking expected properties, as we discussed yesterday during code review, this needs broader discussion. The sole purpose of this user story was to get an end-end working model without breaking the existing functionality. Hence the logic in this implementation mirrored exactly what was in the original sample code.
We may want to address the design as part of the WattTime implementation instead of this class.